### PR TITLE
Add a stage in the gitlab build for artifacts upload

### PR DESCRIPTION
**Pull request by pierotibou**
I've done a few changes:
- Added a stage to publish artifacts
- Renamed stages
- Added credentials
- Added a retry mechanism because it failed on my first attempt with a known flaky error. The retry mechanism itself hasn't been tested, because obviously the copy didn't failed after I've implemented it.

I've finaly understood that I can test the job on a branch by removing the the `only` parameter. This one is working.

@DataDog/apm-dotnet

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -19,10 +20,34 @@ driver_build:
...
+  tags: ["runner:windows-docker", "windowsversion:1809"]
+  dependencies: 
+    - build
+  script:
+    - $result =  aws sts assume-role --role-arn "arn:aws:iam::486234852809:role/ci-datadog-windows-filter" --role-session-name AWSCLI-Session
+    - $resultjson = $result | convertfrom-json
+    - $credentials = $($resultjson.Credentials)
+    - $Env:AWS_ACCESS_KEY_ID="$($credentials.AccessKeyId)"
+    - $Env:AWS_SECRET_ACCESS_KEY="$($credentials.SecretAccessKey)"
+    - $Env:AWS_SESSION_TOKEN="$($credentials.SessionToken)"
```
Comment (andrewlock) [inline .gitlab-ci.yml:44]: Are you sure this is the correct way to set these 🤔 Doesn't this expose them in the logs?

Comment (pierotibou) [reply .gitlab-ci.yml:44]: > correct way

100% sure this is the best way, no, clearly. That said:
- it's used ~often in dd already (cf [codesearch](https://cs.github.com/?scopeName=DataDog&scope=org%3ADataDog&q=aws+sts+assume-role))
- it does not print out the keys (cf [log](https://gitlab.ddbuild.io/DataDog/dd-trace-dotnet/-/jobs/111710362#L34))

**Comment by andrewlock** [comment]
## Benchmarks Report :snail:

Benchmarks for #2351 compared to [master](https://github.com/DataDog/dd-trace-dotnet/tree/cae22b3e142ce7b7109fe5449af74bf3bf7fb080): 
* 1 benchmarks are **faster**, with geometric mean 1.172
* 1 benchmarks are **slower**, with geometric mean 1.223
* All benchmarks have **the same** allocations


The following thresholds were used for comparing the benchmark speeds:
* Mann–Whitney U test with statistical test for significance of **5%**
* Only results indicating a difference greater than **10%** and **0.3 ns** are considered.

Allocation changes below **0.5%** are ignored.


### Benchmark details


<details><summary><strong>Benchmarks.Trace.AgentWriterBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/cae22b3e142ce7b7109fe5449af74bf3bf7fb080)|`WriteAndFlushEnrichedTraces`|net472|861μs|4.46μs|20.4μs|0| 0|0|3.16&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/cae22b3e142ce7b7109fe5449af74bf3bf7fb080)|`WriteAndFlushEnrichedTraces`|netcoreapp3.1|698μs|1.86μs|6.72μs|0| 0|0|2.57&nbsp;KB|
|#2351|`WriteAndFlushEnrichedTraces`|net472|858μs|2.02μs|7.56μs|0| 0|0|3.16&nbsp;KB|
|#2351|`WriteAndFlushEnrichedTraces`|netcoreapp3.1|693μs|3.77μs|22.9μs|0| 0|0|2.57&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.AspNetCoreBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/cae22b3e142ce7b7109fe5449af74bf3bf7fb080)|`SendRequest`|net472|0ns|0ns|0ns|0| 0|0|0&nbsp;b|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/cae22b3e142ce7b7109fe5449af74bf3bf7fb080)|`SendRequest`|netcoreapp3.1|298μs|1.41μs|5.82μs|0.145| 0|0|19.67&nbsp;KB|
|#2351|`SendRequest`|net472|0ns|0ns|0ns|0| 0|0|0&nbsp;b|
|#2351|`SendRequest`|netcoreapp3.1|301μs|1.3μs|4.86μs|0.148| 0|0|19.67&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.DbCommandBenchmark</strong> - Faster :tada: Same allocations :heavy_check_mark:



#### Faster :tada: in #2351

|  Benchmark         | base/diff | Base Median (ns) | Diff Median (ns) | Modality   |
|:----------|-----------:|-----------:|--------:|--------:|
| Benchmarks.Trace.DbCommandBenchmark.ExecuteNonQuery&#8209;net472 | 1.172 | 1,871.22 | 1,597.25 | 

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/cae22b3e142ce7b7109fe5449af74bf3bf7fb080)|`ExecuteNonQuery`|net472|1.87μs|5.94ns|21.4ns|0.0929| 0.000911|0|586&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/cae22b3e142ce7b7109fe5449af74bf3bf7fb080)|`ExecuteNonQuery`|netcoreapp3.1|1.5μs|7.69ns|45.5ns|0.00845| 0|0|624&nbsp;B|
|#2351|`ExecuteNonQuery`|net472|1.59μs|8.06ns|36.9ns|0.0928| 0.000814|0|586&nbsp;B|
|#2351|`ExecuteNonQuery`|netcoreapp3.1|1.51μs|7.18ns|32.1ns|0.00875| 0|0|624&nbsp;B|
</details>


<details><summary><strong>Benchmarks.Trace.ElasticsearchBenchmark</strong> - Slower :warning: Same allocations :heavy_check_mark:



#### Slower :warning: in #2351

|  Benchmark         | diff/base | Base Median (ns) | Diff Median (ns) | Modality   |
|:----------|-----------:|-----------:|--------:|--------:|
| Benchmarks.Trace.ElasticsearchBenchmark.CallElasticsearchAsync&#8209;net472 | 1.223 | 2,473.36 | 3,025.87 | 

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/cae22b3e142ce7b7109fe5449af74bf3bf7fb080)|`CallElasticsearch`|net472|2.46μs|9.35ns|38.5ns|0.124| 0|0|794&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/cae22b3e142ce7b7109fe5449af74bf3bf7fb080)|`CallElasticsearch`|netcoreapp3.1|1.64μs|7.82ns|35.8ns|0.0114| 0|0|784&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/cae22b3e142ce7b7109fe5449af74bf3bf7fb080)|`CallElasticsearchAsync`|net472|2.49μs|8.05ns|30.1ns|0.146| 0|0|931&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/cae22b3e142ce7b7109fe5449af74bf3bf7fb080)|`CallElasticsearchAsync`|netcoreapp3.1|1.77μs|8.02ns|31.1ns|0.0124| 0|0|904&nbsp;B|
|#2351|`CallElasticsearch`|net472|2.3μs|12.1ns|58ns|0.124| 0|0|794&nbsp;B|
|#2351|`CallElasticsearch`|netcoreapp3.1|1.64μs|6.83ns|26.5ns|0.0107| 0|0|784&nbsp;B|
|#2351|`CallElasticsearchAsync`|net472|3.02μs|14.9ns|63.3ns|0.145| 0|0|931&nbsp;B|
|#2351|`CallElasticsearchAsync`|netcoreapp3.1|1.7μs|6.24ns|24.2ns|0.0121| 0|0|904&nbsp;B|
</details>


<details><summary><strong>Benchmarks.Trace.GraphQLBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/cae22b3e142ce7b7109fe5449af74bf3bf7fb080)|`ExecuteAsync`|net472|2.69μs|10.1ns|37.9ns|0.164| 0|0|1.05&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/cae22b3e142ce7b7109fe5449af74bf3bf7fb080)|`ExecuteAsync`|netcoreapp3.1|1.94μs|8.68ns|32.5ns|0.0139| 0|0|1.02&nbsp;KB|
|#2351|`ExecuteAsync`|net472|2.73μs|15.1ns|90.5ns|0.165| 0|0|1.05&nbsp;KB|
|#2351|`ExecuteAsync`|netcoreapp3.1|1.95μs|6.24ns|23.3ns|0.0146| 0|0|1.02&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.HttpClientBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/cae22b3e142ce7b7109fe5449af74bf3bf7fb080)|`SendAsync`|net472|6.16μs|29.5ns|121ns|0.351| 0|0|2.26&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/cae22b3e142ce7b7109fe5449af74bf3bf7fb080)|`SendAsync`|netcoreapp3.1|4.36μs|17.1ns|66.4ns|0.0306| 0|0|2.16&nbsp;KB|
|#2351|`SendAsync`|net472|6.17μs|33.6ns|181ns|0.354| 0|0|2.26&nbsp;KB|
|#2351|`SendAsync`|netcoreapp3.1|4.39μs|23ns|147ns|0.0284| 0|0|2.16&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.ILoggerBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/cae22b3e142ce7b7109fe5449af74bf3bf7fb080)|`EnrichedLog`|net472|3.47μs|11.1ns|41.4ns|0.225| 0|0|1.44&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/cae22b3e142ce7b7109fe5449af74bf3bf7fb080)|`EnrichedLog`|netcoreapp3.1|3.17μs|15.8ns|68.7ns|0.0215| 0|0|1.52&nbsp;KB|
|#2351|`EnrichedLog`|net472|3.48μs|16.9ns|73.7ns|0.224| 0|0|1.44&nbsp;KB|
|#2351|`EnrichedLog`|netcoreapp3.1|3.23μs|15.2ns|74.6ns|0.0213| 0|0|1.52&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.Log4netBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/cae22b3e142ce7b7109fe5449af74bf3bf7fb080)|`EnrichedLog`|net472|294μs|1.04μs|3.6μs|0.434| 0.145|0|4.31&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/cae22b3e142ce7b7109fe5449af74bf3bf7fb080)|`EnrichedLog`|netcoreapp3.1|235μs|866ns|3.35μs|0| 0|0|4.19&nbsp;KB|
|#2351|`EnrichedLog`|net472|290μs|717ns|2.78μs|0.306| 0.153|0|4.31&nbsp;KB|
|#2351|`EnrichedLog`|netcoreapp3.1|239μs|1.18μs|4.88μs|0| 0|0|4.19&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.NLogBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/cae22b3e142ce7b7109fe5449af74bf3bf7fb080)|`EnrichedLog`|net472|8.13μs|37.8ns|239ns|0.502| 0|0|3.21&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/cae22b3e142ce7b7109fe5449af74bf3bf7fb080)|`EnrichedLog`|netcoreapp3.1|6.61μs|36.2ns|208ns|0.049| 0|0|3.58&nbsp;KB|
|#2351|`EnrichedLog`|net472|8.09μs|29.7ns|115ns|0.503| 0|0|3.21&nbsp;KB|
|#2351|`EnrichedLog`|netcoreapp3.1|6.4μs|25.5ns|98.8ns|0.0505| 0|0|3.58&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.RedisBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/cae22b3e142ce7b7109fe5449af74bf3bf7fb080)|`SendReceive`|net472|2.16μs|8.62ns|33.4ns|0.158| 0|0|1&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/cae22b3e142ce7b7109fe5449af74bf3bf7fb080)|`SendReceive`|netcoreapp3.1|2.12μs|11ns|53.7ns|0.0137| 0|0|1&nbsp;KB|
|#2351|`SendReceive`|net472|2.33μs|11.2ns|43.4ns|0.157| 0|0|1&nbsp;KB|
|#2351|`SendReceive`|netcoreapp3.1|2.01μs|9.64ns|37.3ns|0.0141| 0|0|1&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.SerilogBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/cae22b3e142ce7b7109fe5449af74bf3bf7fb080)|`EnrichedLog`|net472|5.96μs|27.8ns|115ns|0.29| 0|0|1.85&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/cae22b3e142ce7b7109fe5449af74bf3bf7fb080)|`EnrichedLog`|netcoreapp3.1|5.2μs|23.4ns|87.7ns|0.0204| 0|0|1.47&nbsp;KB|
|#2351|`EnrichedLog`|net472|6.54μs|32.8ns|143ns|0.289| 0|0|1.85&nbsp;KB|
|#2351|`EnrichedLog`|netcoreapp3.1|5.22μs|17.2ns|64.5ns|0.0206| 0|0|1.47&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.SpanBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/cae22b3e142ce7b7109fe5449af74bf3bf7fb080)|`StartFinishSpan`|net472|935ns|3.53ns|12.7ns|0.0704| 0|0|449&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/cae22b3e142ce7b7109fe5449af74bf3bf7fb080)|`StartFinishSpan`|netcoreapp3.1|876ns|4.17ns|16.2ns|0.00623| 0|0|448&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/cae22b3e142ce7b7109fe5449af74bf3bf7fb080)|`StartFinishScope`|net472|1.12μs|4.2ns|16.2ns|0.0828| 0|0|530&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/cae22b3e142ce7b7109fe5449af74bf3bf7fb080)|`StartFinishScope`|netcoreapp3.1|1.05μs|4.91ns|29.4ns|0.00781| 0|0|568&nbsp;B|
|#2351|`StartFinishSpan`|net472|914ns|3.28ns|11.8ns|0.0703| 0|0|449&nbsp;B|
|#2351|`StartFinishSpan`|netcoreapp3.1|895ns|3.46ns|13.4ns|0.00652| 0|0|448&nbsp;B|
|#2351|`StartFinishScope`|net472|1.16μs|6.48ns|39.4ns|0.0828| 0|0|530&nbsp;B|
|#2351|`StartFinishScope`|netcoreapp3.1|1.06μs|2.74ns|10.2ns|0.00785| 0|0|568&nbsp;B|
</details>

**Comment by andrewlock** [comment]
## Code Coverage Report :bar_chart:

:heavy_check_mark: Merging #2351 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/872a15956cf369773d48b25c7eab644b08d74701) will **not change** line coverage
:heavy_check_mark: Merging #2351 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/872a15956cf369773d48b25c7eab644b08d74701) will **not change** branch coverage
:no_entry: Merging #2351 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/872a15956cf369773d48b25c7eab644b08d74701) will will **increase** complexity by `42`


|           | [master](https://github.com/DataDog/dd-trace-dotnet/tree/872a15956cf369773d48b25c7eab644b08d74701) | #2351       | Change   | 
|:----------|:-----------:|:-----------:|:--------:|
| Lines     | `10429` / `14678` | `10452` / `14767` |          |
| Lines %   | `71%`      | `71%`      |  `0%` :heavy_check_mark:  |
| Branches  | `5361` / `7739` | `5371` / `7765` |          |
| Branches %| `69%`      | `69%`      |  `0%` :heavy_check_mark:  |
| Complexity|   `8850`      | `8892`        |  `42`  :no_entry:    |

View the full report for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=88012&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/88012/artifacts?artifactName=Code%20Coverage%20Report_88012&api-version=6.0&%24format=zip)  
* [HTML report for this PR #2351](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=88095&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/88095/artifacts?artifactName=Code%20Coverage%20Report_88095&api-version=6.0&%24format=zip)

### Datadog.Trace Breakdown :heavy_check_mark:

|        | [master](https://github.com/DataDog/dd-trace-dotnet/tree/872a15956cf369773d48b25c7eab644b08d74701) | #2351       | Change   | 
|:-------|:-----------:|:-----------:|:--------:|
| Lines %| `71%`      | `71%`       |  `0%` :heavy_check_mark:  |
| Branches %| `69%`      | `69%`       |  `0%` :heavy_check_mark:  |
| Complexity| `8850`      | `8892`       |  `42` :no_entry:  |

The following classes have significant coverage changes.

| File    | Line coverage change | Branch coverage change | Complexity change |
|:--------|:--------------------:|:----------:|:--------:|
| [Datadog.Trace.ClrProfiler.AutoInstrumentation.Couchbase.IIOServiceExecuteIntegrationBis](https://github.com/DataDog/dd-trace-dotnet/tree/d98a4604b05696afd571bea9c86c28735eb9b009/tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Couchbase/IIOServiceExecuteIntegrationBis.cs) | `-100%` :no_entry: | `0%` :heavy_check_mark:   | `0` :heavy_check_mark: |
| [Datadog.Trace.ClrProfiler.AutoInstrumentation.Testing.MsTestV2.TestMethodRunnerExecuteIntegration](https://github.com/DataDog/dd-trace-dotnet/tree/d98a4604b05696afd571bea9c86c28735eb9b009/tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Testing/MsTestV2/TestMethodRunnerExecuteIntegration.cs) | `-33%` :no_entry: | `-19%` :no_entry:   | `0` :heavy_check_mark: |
| [Datadog.Trace.ClrProfiler.AutoInstrumentation.Testing.NUnit.NUnitCompositeWorkItemSkipChildrenIntegration](https://github.com/DataDog/dd-trace-dotnet/tree/d98a4604b05696afd571bea9c86c28735eb9b009/tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Testing/NUnit/NUnitCompositeWorkItemSkipChildrenIntegration.cs) | `-8%` :no_entry: | `-8%` :no_entry:   | `0` :heavy_check_mark: |
| [Datadog.Trace.Agent.StreamFactories.UnixDomainSocketStreamFactory](https://github.com/DataDog/dd-trace-dotnet/tree/d98a4604b05696afd571bea9c86c28735eb9b009/tracer/src/Datadog.Trace/Agent/StreamFactories/UnixDomainSocketStreamFactory.cs) | `8%` :heavy_check_mark: | `0%` :heavy_check_mark:   | `0` :heavy_check_mark: |
| [Datadog.Trace.Agent.Transports.HttpStreamRequestFactory](https://github.com/DataDog/dd-trace-dotnet/tree/d98a4604b05696afd571bea9c86c28735eb9b009/tracer/src/Datadog.Trace/Agent/Transports/HttpStreamRequestFactory.cs) | `20%` :heavy_check_mark: | `0%` :heavy_check_mark:   | `0` :heavy_check_mark: |


The following classes were added in #2351:

| File    | Line coverage | Branch coverage | Complexity |
|:--------|:--------------------:|:----------:|:--------:|
| [Datadog.Trace.ClrProfiler.ServerlessInstrumentation.AWS.LambdaCommon](https://github.com/DataDog/dd-trace-dotnet/pull/2351/files) | `0%` | `0%` | `5` |
| [Datadog.Trace.ClrProfiler.ServerlessInstrumentation.AWS.LambdaNoParam](https://github.com/DataDog/dd-trace-dotnet/pull/2351/files) | `0%` | `0%` | `4` |
| [Datadog.Trace.ClrProfiler.ServerlessInstrumentation.AWS.LambdaOneParam](https://github.com/DataDog/dd-trace-dotnet/pull/2351/files) | `0%` | `0%` | `4` |
| [Datadog.Trace.ClrProfiler.ServerlessInstrumentation.AWS.LambdaTwoParams](https://github.com/DataDog/dd-trace-dotnet/pull/2351/files) | `0%` | `0%` | `4` |
| [Datadog.Trace.ClrProfiler.ServerlessInstrumentation.LambdaHandler](https://github.com/DataDog/dd-trace-dotnet/pull/2351/files) | `100%` | `100%` | `8` |
| ...And 1 more  | | | |


View the full reports for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=88012&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/88012/artifacts?artifactName=Code%20Coverage%20Report_88012&api-version=6.0&%24format=zip)  
* [HTML report for this PR #2351](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=88095&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/88095/artifacts?artifactName=Code%20Coverage%20Report_88095&api-version=6.0&%24format=zip)

**Review by andrewlock — APPROVED** [review]

**Pull request merged**